plugins/destination/duckdb/client/read.go

```diff
@@ -82,170 +81,109 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 }
 
 func convertToSingleRowRecords(sc *arrow.Schema, rec arrow.Record) []arrow.Record {
-	records := make([]arrow.Record, rec.NumRows())
-	for i := int64(0); i < rec.NumRows(); i++ {
-		records[i] = reverseTransformRecord(sc, rec.NewSlice(i, i+1))
+	// transform arrays first
+	transformed := reverseTransformRecord(sc, rec)
```

Comment:
Can you explain the purpose of this refactor?

Reply:
If we get a record with several rows, we'll have much less function calls as a result.
In the previous implementation we would transform the data row by row, calling at least `rec.NumCols()` functions for each row.
Now we are calling the transformation first, resulting in a multi-row record being returned, and then we just slice it.